src/shared/telemetry/telemetryLogger.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MetricDatum, MetadataEntry } from './clienttelemetry'
+
+interface MetadataQuery {
+    /** Metric name to look up in the log */
+    readonly metricName: string
+    /** Returns the metric containing the metadata instead of just the metadata (default: false) */
+    readonly returnMetric?: boolean
+    /** Attributes to filter out of the metadata */
+    readonly filters?: string[]
+}
+
+interface MetricQuery extends MetadataQuery {
+    readonly returnMetric: true
+}
+
+interface Metadata {
+    [key: string]: string | undefined
+}
+
+export type Query = MetricQuery | MetadataQuery
```

Comment:
Unnecessary export. Is the type even necessary?

Reply:
Probably not. I was using the unions since I did have more fields.